drivers/soundwire/bus.c

```diff
@@ -1264,15 +1272,30 @@ static int sdw_initialize_slave(struct sdw_slave *slave)
 	if (slave->bus->prop.quirks & SDW_MASTER_QUIRKS_CLEAR_INITIAL_CLASH) {
 		/* Clear bus clash interrupt before enabling interrupt mask */
 		status = sdw_read_no_pm(slave, SDW_SCP_INT1);
+		if (status < 0) {
+			dev_err(&slave->dev,
+				"SDW_SCP_INT1 read failed:%d\n", status);
+			return status;
+		}
 		if (status & SDW_SCP_INT1_BUS_CLASH) {
 			dev_warn(&slave->dev, "Bus clash detected before INT mask is enabled\n");
-			sdw_write_no_pm(slave, SDW_SCP_INT1, SDW_SCP_INT1_BUS_CLASH);
+			ret = sdw_write_no_pm(slave, SDW_SCP_INT1, SDW_SCP_INT1_BUS_CLASH);
+			if (ret < 0) {
+				dev_err(&slave->dev,
+					"SDW_SCP_INT1 write failed:%d\n", ret);
+				return ret;
+			}
 		}
 	}
 	if ((slave->bus->prop.quirks & SDW_MASTER_QUIRKS_CLEAR_INITIAL_PARITY) &&
 	    !(slave->prop.quirks & SDW_SLAVE_QUIRKS_INVALID_INITIAL_PARITY)) {
 		/* Clear parity interrupt before enabling interrupt mask */
 		status = sdw_read_no_pm(slave, SDW_SCP_INT1);
+		if (status < 0) {
+			dev_err(&slave->dev,
+				"SDW_SCP_INT1 read failed:%d\n", status);
+			return status;
+		}
 		if (status & SDW_SCP_INT1_PARITY) {
 			dev_warn(&slave->dev, "PARITY error detected before INT mask is enabled\n");
 			sdw_write_no_pm(slave, SDW_SCP_INT1, SDW_SCP_INT1_PARITY);
```

Comment:
L1301, check ret value...?

Reply:
yes, I forgot to push this is a different commit.